docs/OnTrack/Numbas/ProjectSignOffNumbas.md

```diff
@@ -0,0 +1,71 @@
+# Project Name: \<\<Project Numbas Integration\>\>
+
+## Delivery Lead: \<\<Delivery Daniel Maddern\>\>
+
+## Student Name:\<\<Student Daniel Maddern\>\>
+
+## Clients: \<\< Julien, Andrew, Jake \>\>
+
+## **Scope**
+
+The purpose of this project is to look at how we can integrate Numbas testing into Ontrack. With the
+aim for a test to be presented to the student on submission of a task, prior to submitting
+reflections or other required documents. The aim is to be able to let the Unit Chair import, setup
+and assign tests to the tasks. Also for students to be able to complete tests as part of the
+submission process.
+
+## **Outcomes**
+
+This project will be deliverying a project scope for the clients to sign off on.
+
+A feasiblity study of the ways this can be implemetned.
+
+Rough hand drawn design documents for how this intergration will work.
+
+A data flow diagram of how different data will be accesssed and encapsualted.
+
+Backend coding changes to accomodate and store the tests.
+
+Front end code to support the changes from the Unit chair and student view.
+
+## **Delivery**
+
+There will need to be a means to upload the test files that are created locally via Numbas. An
+Addtional window after "requesting feedback" on a task that will present the test, this will then
+either take you to the next stage if you pass or go back to the task screen if you do not pass.
+
+There will need to be a configuration section within the Unit chair task setup page.
+
+We will provide different options for the test setup such as: 1: Restricted test attempts 2:
```

Comment:
It would be good to capture this as two separate options:

1. Restrict number of attempts before requiring tutor reset (set number or unlimited)
1. Require delay between attempts (set minutes, increasing minutes between successive fails)

This then also adds a feature to reset attempts.